library/core/src/any.rs

```diff
@@ -663,12 +663,20 @@ impl dyn Any + Send + Sync {
 /// noting that the hashes and ordering will vary between Rust releases. Beware
 /// of relying on them inside of your code!
 #[derive(Clone, Copy, Debug, Hash, Eq)]
-#[derive_const(PartialEq, PartialOrd, Ord)]
+#[derive_const(PartialOrd, Ord)]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct TypeId {
     t: u64,
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_const_unstable(feature = "const_cmp_type_id", issue = "101871")]
+impl const PartialEq for TypeId {
+    fn eq(&self, other: &Self) -> bool {
+        self.t.eq(&other.t)
+    }
```

Comment:
```suggestion
    #[inline]
    fn eq(&self, other: &Self) -> bool {
        self.t == other.t
    }
```

This produces less IR and allows the compiler to "see through" this function